cmd/gitserver/server/patch.go

```diff
@@ -564,6 +547,53 @@ func (s *Server) shelveChangelist(ctx context.Context, req protocol.CreateCommit
 	return cid, nil
 }
 
+type gitCommand struct {
+	ctx        context.Context
+	workingDir string
+	env        []string
+}
+
+func (g gitCommand) commandContext(args ...string) *exec.Cmd {
+	cmd := exec.CommandContext(g.ctx, "git", args...)
+	cmd.Dir = g.workingDir
+	cmd.Env = g.env
+	return cmd
+}
+
+func (g gitCommand) getChangelistIdFromCommit(baseCommit string) (string, error) {
```

Comment:
Nit: In Go the usual convention is to use uppercase for abbreviations in names. :) 
(call sites will need to be updated as I didn't add suggestion for those)

```suggestion
func (g gitCommand) getChangelistIDFromCommit(baseCommit string) (string, error) {
```